### DISCUSSION TITLE
Should I make a three.js extension?

### DISCUSSION THREAD

**Discussion by Brackets-Coder** [TurboWarp/extensions#1754] [Ideas]
I'm aware that Simple3D exists and it can do quite a lot but it'd be great if we could have a more advanced 3D extension with support for custom shaders, collision/rigid body physics, PBR and advanced lighting such as volumetrics and dynamic shadows.

Of course, this would be a very advanced and complicated extension that would easily take several months or even years to develop. I just wanna see what you guys think.

I couldn't start development anyway until I finish #1732.

**Comment by SharkPool-SP** [comment]
There's CST 3d that uses three js

**Comment by SharkPool-SP** [comment]
https://raw.githubusercontent.com/CST1229/turbowarp-extensions/3d/extensions/CST1229/3d.js

**Reply by CST1229** [reply]
that has barely anything in it (though someone else is working on another 3d extension called TO3D)

**Comment by CubesterYT** [comment]
It's anyones game with three.js. I'd say  go for it, but keep in mind there's like 5 other people doing the same, with different implementations.

**Reply by Brackets-Coder** [reply]
Good to know. I won't spend extra time if there's numerous other developers working on a similar extension that I wasn't aware of.

**Comment by Brackets-Coder** [comment]
Nevermind it's sorted out now
